src/lib/y2partitioner/dialogs/btrfs_subvolume.rb

```diff
@@ -0,0 +1,192 @@
+require "yast"
+require "cwm"
+require "y2partitioner/dialogs/popup"
+
+Yast.import "Popup"
+
+module Y2Partitioner
+  module Dialogs
+    # Popup dialog to create a btrfs subvolume
+    class BtrfsSubvolume < Popup
+      attr_reader :filesystem
+      attr_reader :form
+
+      # @param filesystem [Y2Storage::Filesystems::BlkFilesystem] a btrfs filesystem
+      def initialize(filesystem)
+        textdomain "storage"
+
+        @filesystem = filesystem
+        @form = Form.new
+      end
+
+      def title
+        _("Add subvolume")
+      end
+
+      # Shows widgets for the subvolume attributes
+      def contents
+        HVSquash(
+          VBox(
+            Left(SubvolumePath.new(form, filesystem: filesystem)),
+            Left(SubvolumeNocow.new(form, filesystem: filesystem))
+          )
+        )
+      end
+
+      # Custom layout
+      #
+      # Similar to {Popup#layout} but without help button
+      def layout
+        VBox(
+          HSpacing(50),
+          Left(Heading(Id(:title), title)),
+          VStretch(),
+          VSpacing(1),
+          VBox(ReplacePoint(Id(:contents), Empty())),
+          VSpacing(1),
+          VStretch(),
+          ButtonBox(
+            PushButton(Id(:ok), Opt(:default), Yast::Label.OKButton),
+            PushButton(Id(:cancel), Yast::Label.CancelButton)
+          )
+        )
+      end
+
+      # Form object for the dialog
+      #
+      # Widgets use this object to storage data
+      class Form
```

Comment:
I suggest to use slightly longer data, that is more obvious. Like `SubvolumeForm`. Form standalone is too ambitious name